src/imf_reader/sdr/read_interest_rate.py

```diff
@@ -0,0 +1,105 @@
+"""Module to read SDR interest and exchange rates from the IMF website
+
+"""
+
+import requests
+import pandas as pd
+import io
+from functools import lru_cache
+
+from imf_reader.config import logger
+
+
+BASE_URL: str = "https://www.imf.org/external/np/fin/data/sdr_ir.aspx"
+
+
+def read_data():
+    """Read the data from the IMF website"""
+
+    data = {
+        "__EVENTTARGET": "lbnTSV",
+    }
+
+    try:
+        response = requests.post(BASE_URL, data=data)
+        response.raise_for_status()
+
+    except requests.exceptions.RequestException as e:
+        raise ConnectionError(f"Could not connect to {BASE_URL}. Error: {str(e)}")
+
+    try:
+        return pd.read_csv(
+            io.BytesIO(response.content), delimiter="/t", engine="python"
+        )
+
+    except pd.errors.ParserError as e:
+        raise ValueError(f"Could not parse data. Error: {str(e)}")
+
+
+def clean_data(df: pd.DataFrame) -> pd.DataFrame:
+    """Cleaning/parsing steps for the data. split tab separated value into separate columns, rename columns, assign types, and additional formatting"""
+    columns = {
+        "Effective from": "effective_from",
+        "Effective to": "effective_to",
+    }
+
+    df = df.iloc[:, 0].str.split("\t", expand=True)
+    df.columns = df.iloc[0]
+    df = df.iloc[1:]
+
+    return (
+        df.rename(columns=columns)
+        .loc[:, columns.values()]
+        .dropna(subset=["effective_to"])
+        .pipe(_format_data)
+        .assign(
+            interest_rate=lambda d: pd.to_numeric(d.interest_rate, errors="coerce"),
+            effective_from=lambda d: pd.to_datetime(d.effective_from),
+            effective_to=lambda d: pd.to_datetime(d.effective_to),
+        )
+    )
+
+
+def _format_data(df: pd.DataFrame) -> pd.DataFrame:
```

Comment:
this formats and filters. It would help to split in two. Also because clean data does quite a bit - and this one also deals with cleaning)